piptools/utils.py

```diff
@@ -101,15 +104,28 @@ def format_requirement(
     ireq: InstallRequirement,
     marker: Optional[Marker] = None,
     hashes: Optional[Set[str]] = None,
+    from_dir: Optional[str] = None,
 ) -> str:
     """
     Generic formatter for pretty printing InstallRequirements to the terminal
     in a less verbose way than using its `__str__` method.
     """
-    if ireq.editable:
-        line = f"-e {ireq.link.url}"
-    elif is_url_requirement(ireq):
-        line = ireq.link.url
+    if is_url_requirement(ireq):
+        if ireq.link.is_file:
+            if from_dir:
+                path = "file:" + os.path.relpath(
+                    ireq.local_file_path, from_dir
+                ).replace(os.path.sep, "/")
+            else:
+                path = path_to_url(ireq.local_file_path)
+            egg = f"#egg={ireq.link.egg_fragment}" if ireq.link.egg_fragment else ""
```

Comment:
```suggestion
            egg = f"#egg={ireq.link.egg_fragment}" if ireq.link.egg_fragment is not None else ""
```

Minor: I would check for `None` explicitly.

Reply:
:+1:  5afd75e113acb3720bc4be07f14c01720cb81d79